app/app.go

```diff
@@ -180,31 +183,31 @@ func NewHeimdallApp(logger log.Logger, db dbm.DB, baseAppOptions ...func(*bam.Ba
 		authTypes.StoreKey,
 		bankTypes.StoreKey,
 		supplyTypes.StoreKey,
-		// gov.StoreKey,
+		govTypes.StoreKey,
 		stakingTypes.StoreKey,
 		checkpointTypes.StoreKey,
 		borTypes.StoreKey,
 		clerkTypes.StoreKey,
-		params.StoreKey,
 		topupTypes.StoreKey,
+		paramsTypes.StoreKey,
 	)
-	tkeys := sdk.NewTransientStoreKeys(params.TStoreKey)
+	tkeys := sdk.NewTransientStoreKeys(paramsTypes.TStoreKey)
 
 	// create heimdall app
 	var app = &HeimdallApp{
 		cdc:       cdc,
 		BaseApp:   bApp,
 		keys:      keys,
 		tkeys:     tkeys,
-		subspaces: make(map[string]params.Subspace),
+		subspaces: make(map[string]subspace.Subspace),
 	}
 
 	// init params keeper and subspaces
-	app.ParamsKeeper = params.NewKeeper(app.cdc, keys[params.StoreKey], tkeys[params.TStoreKey], params.DefaultCodespace)
+	app.ParamsKeeper = params.NewKeeper(app.cdc, keys[paramsTypes.StoreKey], tkeys[paramsTypes.TStoreKey], paramsTypes.DefaultCodespace)
 	app.subspaces[authTypes.ModuleName] = app.ParamsKeeper.Subspace(authTypes.DefaultParamspace)
 	app.subspaces[bankTypes.ModuleName] = app.ParamsKeeper.Subspace(bankTypes.DefaultParamspace)
 	app.subspaces[supplyTypes.ModuleName] = app.ParamsKeeper.Subspace(supplyTypes.DefaultParamspace)
-	// app.subspaces[gov.ModuleName] = app.ParamsKeeper.Subspace(gov.DefaultParamspace).WithKeyTable(gov.ParamKeyTable())
+	app.subspaces[govTypes.ModuleName] = app.ParamsKeeper.Subspace(govTypes.DefaultParamspace).WithKeyTable(govTypes.ParamKeyTable())
 	app.subspaces[stakingTypes.ModuleName] = app.ParamsKeeper.Subspace(stakingTypes.DefaultParamspace)
```

Comment:
later when we are creating staking keeper at line no 248, we are passing **app.subspaces[stakingTypes.ModuleName]** as argument. 

Inside keeper constructor, we are setting **paramSpace:  paramSpace.WithKeyTable(types.ParamKeyTable())**
WithKeyTable function returns a new subspace with modified table adding staking params. 
so there will be mismatch of staking subspace in paramskeeper and stakingkeeper 

so we should store pointer to the subspace i think in the map in app.go.

Reply:
Also do we need to have subspace for module(eg...staking) both in stakingkeeper and paramskeeper??

Reply:
That is modifying the same subspace with `WithKeyTable`